### PR TITLE
⚡️ V2 Rewrite — Get rid of lz4 dependency

### PR DESCRIPTION
**Pull request by paambaati**
The [`lz4`](https://github.com/pierrec/node-lz4) package has installation issues, and seems to be not maintained anymore.

This PR aims to remove this dependency entirely and instead use [`lz4-asm`](https://www.npmjs.com/package/lz4-asm). It is actively maintained and offers near-native performance.

This PR should fix https://github.com/indix/kafkajs-lz4/issues/5 and https://github.com/indix/kafkajs-lz4/issues/3

Other changes include —

1. Minimum required Node.js version now become 10 (because of `lz4-asm`).
2. Travis CI configuration is removed in favour of Github Actions.
3. TSLint is replaced with ESLint + Prettier for a more modern and consistent style.
4. Test suite now runs against Kafka v2.5.0.
5. All dependencies upgraded to their latest stable versions.

**Rollout strategy**

We'll be publishing this as a a beta (`yarn add kafkajs-lz4@2.0.0-beta.0`), collect feedback and then promote `2.0.0` to stable after a few weeks.

@Prsna23 Can you also upgrade to this internally and ensure this is stable?

CC: @ankon @Freakachoo

### CODE REVIEW

**Comment by paambaati** [comment]
I've just published this to NPM. This is now available to install with —

```
yarn add kafkajs-lz4@2.0.0-beta.0
```

**Comment by Prsna23** [comment]
@paambaati I don't think we will be upgrading it for a while now. I am able to pull this package without a problem.

File: src/index.ts [hunk 1]
```diff
@@ -1,71 +1,87 @@
...
     private async decompress(buffer: Buffer): Promise<Buffer> {
-        return await new Promise<Buffer>(resolve => {
-            const decompressedBuffer = decode(buffer);
-            return resolve(decompressedBuffer);
+        return await new Promise<Buffer>((resolve) => {
+            return lz4Ready.then((lz4) => {
+                const lz4js = lz4.lz4js;
+                const decompressedBuffer: Buffer = lz4js.decompress(buffer);
+                return resolve(decompressedBuffer);
+            });
```
Comment (ankon) [inline src/index.ts:84]: This is still overly complicated.

```
private async decompress(buffer: Buffer): Promise<Buffer> {
  const { lz4js } = await lz4Ready;
  return lz4js.decompress(buffer);
}
```

Same for the compress :)

**Comment by ankon** [comment]
We replaced kafkajs-lz4 with a simple trivial wrapper based on #2. I'll have to check whether switching to lz4-asm would be worth the effort for us, but it's definitely nice to see activity here!

File: src/index.ts (hunk 1 above)
Comment (paambaati) [reply src/index.ts:84]: @ankon Thanks, this has been addressed in https://github.com/indix/kafkajs-lz4/commit/5efecce6fc8e746d43bc21298e169eb62eb65b5a. I've added you as a co-author for that commit!

**Pull request merged**